Heap/Heap.swift

```diff
@@ -89,69 +86,79 @@ public struct Heap<T> {
    * value (for a min-heap).
    */
   public func peek() -> T? {
-    return elements.first
+    return nodes.first
+  }
+  
+  /** Returns the node at given index */
+  public func node(at i: Int) -> T? {
```

Comment:
Do you have a use case in mind for this method (besides using it internally for testing)? I don't see a use case for retrieving a value at the middle of a heap.

Reply:
No, I really only used it for testing purposes

Reply:
I suggest taking this out then. Let's think of another way to test without exposing this member function.